virtual-desktop/src/app/plugin-manager/shared/desktop-plugin-definition.ts

```diff
@@ -38,7 +38,12 @@ export class DesktopPluginDefinitionImpl implements MVDHosting.DesktopPluginDefi
 
   getFramework(): string {
     if (this.hasWebContent) {
-      return this.basePlugin.getWebContent().framework;
+      if ('framework' in this.basePlugin.getWebContent()) {
+        return this.basePlugin.getWebContent().framework;
+      } else {
+        console.warn(`Plugin ${this.getIdentifier()} has no framework specified`);
+        return 'unsupported';
+      }
     } else {
       console.warn(`Plugin ${this.getIdentifier()} has no framework specified`);
       return 'unsupported';
```

Comment:
If this were to return N/A, then that would make sense as compared to 'unsupported'. Then, no console warn would be necessary.